### ISSUE TITLE
The difference between the paper and the code for Attention branch

### ISSUE THREAD

**Issue by AkihiroSasabe** [machine-perception-robotics-group/attention_branch_network#10]
論文と実際のコード(baseline modelがResNetのもの)に細かいですが、違いがあるようにみえますので、
よろしければご確認いただけないでしょうか。該当部分は下記の通りです。

models/cifar/resnet.pyの179行目

    self.att = self.sigmoid(self.bn_att3(self.att_conv3(ax)))

のself.att_conv3(ax)は、119行目で

    self.att_conv3  = nn.Conv2d(num_classes, 1, kernel_size=3, padding=1, bias=False)

のようにカーネルサイズが3と定義されておりますが
論文( https://arxiv.org/pdf/1812.10025.pdf ) のFig. 2(a)のAttention branchのAttention mapを出力する方の分岐には、
    1x1 conv., 1
のようにカーネルサイズが1となっております。
![image](https://user-images.githubusercontent.com/34682532/77287573-30e2c780-6d19-11ea-96ac-78eb3f68a05e.png)

私の勘違いであれば申し訳ございません。
お忙しいところ恐縮ですが、ご確認いただければ幸いです。
何卒宜しくお願い致します。

**Comment by Hiroshi-Fukui** [comment]
Sorry, I didn't notice...
The performances at our CVPR paper are used the 3x3 convolution.

**Comment by AkihiroSasabe** [comment]
Thanks for your reply. 
What are your thoughts on  the effect of using the 3x3 convolution layer instead of the 1x1 one?

Also, I find another gap between the paper and the implementation. 
[In the code,](https://github.com/machine-perception-robotics-group/attention_branch_network/blob/master/models/cifar/resnet.py#L177) Batch Normalization (bn_att2) is inserted between 1x1 conv. (att_conv) and ReLU at the attention branch.

```python
ax = self.relu(self.bn_att2(self.att_conv(ax)))
```

![image](https://user-images.githubusercontent.com/34682532/90954704-9ba75300-e4b1-11ea-9a84-f5f87f7ec9c7.png)

**Comment by thirakawa** [comment]
I'll answer your questions on behalf of the first author because he has already graduated and belongs to the other institution.

Q1. Unfortunately, we haven't try to use 1x1 conv. So, we don't have any insights for the effect of 1x1 conv. to generate an attention map.

Q2. We are sorry for the incorrect figure. As you pointed out, the original network contains a BN layer after 1x1 conv. The figure lacks the BN layer.

We published ABN source codes that are used the experiments reported in our CVPR 2019 paper without any modifications except for removing personal informations. Therefore, please see the program for understanding the detailed network architecture. (i.e., GitHub codes are correct.)

**Comment by AkihiroSasabe** [comment]
Thanks for your quick response and congratulations on his PhD.

**Comment by valencebond** [comment]
hi @AkihiroSasabe , can you achieve the same performance in cifar100?

**Comment by schennupati** [comment]
I think the figure is correct.

If Residual blocks have order ['conv', 'bn', 'relu'] we are not able to reproduce results. This made me realize that author might have used ['relu', 'bn', 'conv'] for experiments that have slightly better performance. 

In that case, output from residual block follows, ['bn', 'conv', 'relu', 'conv', 'bn', 'sigmoid'] in the attention branch. I'm trying to reproduce results with this intuition.
